### ISSUE TITLE
Problem running unittest. Import successful?

### ISSUE THREAD

**Issue by SamAct** [FUSED-Wind/fusedwind#80]
Hello kindly look into the log below. Can you tell me why this is happening?
New to this, could have been mistaken while carrying out the installation process.

```
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\anur>cd c:\python27\openmdao-0.7.0

c:\Python27\openmdao-0.7.0>Scripts\activate
(openmdao-0.7.0) c:\Python27\openmdao-0.7.0>git clone https://github.com/FUSED-
ind/fusedwind.git
Cloning into 'fusedwind'...
remote: Counting objects: 3220, done.
remote: Total 3220 (delta 0), reused 0 (delta 0), pack-reused 3220R
Receiving objects: 100% (3220/3220), 7.06 MiB | 1.33 MiB/s, done.

Resolving deltas: 100% (1898/1898), done.

(openmdao-0.7.0) c:\Python27\openmdao-0.7.0>cd fusedwind

(openmdao-0.7.0) c:\Python27\openmdao-0.7.0\fusedwind>plugin install
installing distribution from current directory as a 'develop' egg
running develop
running egg_info
creating src\fusedwind.egg-info
writing requirements to src\fusedwind.egg-info\requires.txt
writing src\fusedwind.egg-info\PKG-INFO
writing top-level names to src\fusedwind.egg-info\top_level.txt
writing dependency_links to src\fusedwind.egg-info\dependency_links.txt
writing manifest file 'src\fusedwind.egg-info\SOURCES.txt'
reading manifest file 'src\fusedwind.egg-info\SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no directories found matching 'src\fusedwind\sphinx_build\html'
writing manifest file 'src\fusedwind.egg-info\SOURCES.txt'
running build_ext
Creating c:\python27\openmdao-0.7.0\lib\site-packages\fusedwind.egg-link (link
o src)
Removing fusedwind 0.1.0 from easy-install.pth file
Adding fusedwind 0.1.0 to easy-install.pth file

Installed c:\python27\openmdao-0.7.0\fusedwind\src

(openmdao-0.7.0) c:\Python27\openmdao-0.7.0\fusedwind>python
Python 2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015, 20:32:19) [MSC v.1500 32 bit
Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import fusedwind
>>> python -m unittest discover 'src/fusedwind/test' 'test_*.py'
  File "<stdin>", line 1
    python -m unittest discover 'src/fusedwind/test' 'test_*.py'
                     ^
SyntaxError: invalid syntax
>>> quit()

(openmdao-0.7.0) c:\Python27\openmdao-0.7.0\fusedwind>python -m unittest discov
r 'src/fusedwind/test' 'test_*.py'
Traceback (most recent call last):
  File "C:\Python27\Lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27\Lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Lib\unittest\__main__.py", line 12, in <module>
    main(module=None)
  File "C:\Python27\Lib\unittest\main.py", line 94, in __init__
    self.parseArgs(argv)
  File "C:\Python27\Lib\unittest\main.py", line 113, in parseArgs
    self._do_discovery(argv[2:])
  File "C:\Python27\Lib\unittest\main.py", line 214, in _do_discovery
    self.test = loader.discover(start_dir, pattern, top_level_dir)
  File "C:\Python27\Lib\unittest\loader.py", line 204, in discover
    raise ImportError('Start directory is not importable: %r' % start_dir)
ImportError: Start directory is not importable: "'src/fusedwind/test'"

(openmdao-0.7.0) c:\Python27\openmdao-0.7.0\fusedwind>
```

**Comment by Kenneth-T-Moore** [comment]
Well, the first thing I see is this

     >>> python -m unittest discover 'src/fusedwind/test' 'test_*.py'

You should call python at the operating-system command prompt, not inside of the python shell.

**Comment by SamAct** [comment]
Yes true, i corrected it next please see after `quit()`
@Kenneth-T-Moore

**Comment by Kenneth-T-Moore** [comment]
You should be fine if you do this:

    python -m unittest discover 'src/fusedwind/' 'test_*.py'

or alternatively put an empty `__init__.py` into the src/fusedwind/test directory.

**Comment by SamAct** [comment]
@Kenneth-T-Moore 
Nope same type of error. is it because its openmdao-0.7.0 and not 0.10.3.2 ?
See below:
```
(openmdao-0.7.0) c:\Python27\openmdao-0.7.0\fusedwind>python -m unittest discove
r 'src/fusedwind/' 'test_*.py'
Traceback (most recent call last):
  File "C:\Python27\Lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27\Lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Lib\unittest\__main__.py", line 12, in <module>
    main(module=None)
  File "C:\Python27\Lib\unittest\main.py", line 94, in __init__
    self.parseArgs(argv)
  File "C:\Python27\Lib\unittest\main.py", line 113, in parseArgs
    self._do_discovery(argv[2:])
  File "C:\Python27\Lib\unittest\main.py", line 214, in _do_discovery
    self.test = loader.discover(start_dir, pattern, top_level_dir)
  File "C:\Python27\Lib\unittest\loader.py", line 204, in discover
    raise ImportError('Start directory is not importable: %r' % start_dir)
ImportError: Start directory is not importable: "'src/fusedwind/'"
```

**Comment by Kenneth-T-Moore** [comment]
Do you have an `__init__.py` in src/fusedwind?

**Comment by SamAct** [comment]
yes

**Comment by Kenneth-T-Moore** [comment]
I played around a bit using unittest discover on some packages, and I was only able to reproduce your error when `__init__.py` was not present, so I don't know what to advise. You could go into that directory and run each python test file manually, though that sounds tedious. Might also check with the FUSED-wind folks and see if they have any recommendations (though I suppose that's what this github issue page is.)

**Comment by SamAct** [comment]
Some of the posts online related to unittest imply to change the `src/fusedwind/test` to a full path. that doesnt work to i tried `C:/python27/test` and `c:/python27/openmdao-0.10.3.2/fusedwind/src/fusedwind/test`. No change. See [this](https://stackoverflow.com/questions/26922501/app-engine-unit-testing-importerror-start-directory-is-not-importable), [this ](https://stackoverflow.com/questions/12350907/unittest-importerror) and [this](https://stackoverflow.com/questions/25142923/django-testing-importerror-start-directory-is-not-importable).
I also update to openmdao-0.10.3.2 just in case.
@rethore @fzahle

**Comment by fzahle** [comment]
Hi @SamAct
Thanks for trying out fusedwind. I think Ken is correct that it's the missing __init__.py that's causing the issues. You could try to navigate to the test directory and run 

    nosetests .

Also note that this code base is not actively developed any longer since it is based on the old version of OpenMDAO.

**Comment by SamAct** [comment]
@fzahle I tried a few times `nosetest .` First attempt gave me this - All 15 test failed with same error. See below.  
```
======================================================================
ERROR: Failure: ImportError (No module named lib.utilities)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\openmdao-0.10.3.2\lib\site-packages\nose-1.3.3-py2.7.egg\nos
e\loader.py", line 414, in loadTestsFromName
    addr.filename, addr.module)
  File "C:\Python27\openmdao-0.10.3.2\lib\site-packages\nose-1.3.3-py2.7.egg\nos
e\importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "C:\Python27\openmdao-0.10.3.2\lib\site-packages\nose-1.3.3-py2.7.egg\nos
e\importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "C:\Python27\openmdao-0.10.3.2\fusedwind\test\test_plantcost.py", line 12
, in <module>
    from fusedwind.lib.utilities import check_gradient_unit_test
ImportError: No module named lib.utilities

----------------------------------------------------------------------
Ran 15 tests in 0.016s

FAILED (errors=15)
```

All other attempts gave this:

```
(openmdao-0.10.3.2) C:\Python27\openmdao-0.10.3.2\fusedwind\src\fusedwind\test>n
osetest .
'nosetest' is not recognized as an internal or external command,
operable program or batch file.
```
The first attempt is slightly different I copied the test folder away from `src/fusewind/` nevertheless repeated attempts converge.
I have rechecked all the folders they have __init__.py empty file in them.

**Comment by fzahle** [comment]
Do you also have the latest version of fusedwind installed (FUSED-Wind/fusedwind-dev)? If so there'll be a name conflict. The two versions of fusedwind should be installed in separate environments).

On 10. jul. 2017, at 10.11, SamAct <notifications@github.com<mailto:notifications@github.com>> wrote:


@fzahle<https://github.com/fzahle> I tried a few times nosetest . First attempt gave me this - All 15 test failed with same error. See below.

======================================================================
ERROR: Failure: ImportError (No module named lib.utilities)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\openmdao-0.10.3.2\lib\site-packages\nose-1.3.3-py2.7.egg\nos
e\loader.py", line 414, in loadTestsFromName
    addr.filename, addr.module)
  File "C:\Python27\openmdao-0.10.3.2\lib\site-packages\nose-1.3.3-py2.7.egg\nos
e\importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "C:\Python27\openmdao-0.10.3.2\lib\site-packages\nose-1.3.3-py2.7.egg\nos
e\importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "C:\Python27\openmdao-0.10.3.2\fusedwind\test\test_plantcost.py", line 12
, in <module>
    from fusedwind.lib.utilities import check_gradient_unit_test
ImportError: No module named lib.utilities

----------------------------------------------------------------------
Ran 15 tests in 0.016s

FAILED (errors=15)


All other attempts gave this:

(openmdao-0.10.3.2) C:\Python27\openmdao-0.10.3.2\fusedwind\src\fusedwind\test>n
osetest .
'nosetest' is not recognized as an internal or external command,
operable program or batch file.


I have rechecked all the folders they have init.py empty file in them.

**Comment by SamAct** [comment]
No I just have one fusedwind installed and one openmdao as well.

**Comment by SamAct** [comment]
Any other ideas?
[fusedwind_log.txt](https://github.com/FUSED-Wind/fusedwind/files/1929499/fusedwind_log.txt)
Attached is the log.
This is new openmdao, and I am getting the same errors as before. Can you check and suggest please?

Without parenthesis all 15 test Failed.

**Comment by MxLvtt** [comment]
Even though it's an obvious mistake, I just wanted to post this in case someone else is in the same situation ^^

For anyone stumbling across this issue where the added `__init__` files did not solve the issue, here's what I did wrong.

I added the suggested command `python3 -m unittest discover 'tests/' 'test_*.py'` to an NPM script. I didn't keep in mind however that my NPM is initialised one directory above my python project's root:
- MyApp/
    + package.json
    + ...
    + **PythonApp**/
        * src/
        * **tests**/
            - \_\_init\_\_.py
            - test_something.py
            
Simply updating the script to `cd PythonApp && python3 -m unittest discover 'tests/' 'test_*.py'` (and of course adding the `__init__` files) fixed the problem. Pretty obvious in hindsight, but took me a minute to figure it out ^^
